c7n/resources/cw.py

```diff
@@ -963,11 +965,21 @@ def process(self, resources):
         filter_pattern = self.data.get('filter_pattern', '')
         destination_arn = self.data.get('destination_arn')
         distribution = self.data.get('distribution', 'ByLogStream')
+        role_arn = self.data.get('role_arn')
 
         for r in resources:
-            client.put_subscription_filter(
-                logGroupName=r['logGroupName'],
-                filterName=filter_name,
-                filterPattern=filter_pattern,
-                destinationArn=destination_arn,
-                distribution=distribution)
+            if role_arn:
+                client.put_subscription_filter(
+                    logGroupName=r['logGroupName'],
```

Comment:
its a style nit, but rather than large duplicate blocks how about just params = dict(), and conditionally add in rolearn if specified. and then put_sub_filter(**params) we use that pattern a few other places and it cuts down on the duplication.

Reply:
got it